[Bug Fix] server not changing in dropdown inside settings screen

aliseyalvi opened the pull request:
### What does this PR?
This PR fixes the issue related to server not changing in dropdown menu when server is tried to change it do not correctly updates the dropdown selection menu
Also fixed a small glitch in default server value when default server is tried to change, it initially shows no selection due to this bug

### Steps to reproduce
Go to Settings -> Server -> change the server

### Issue number
https://discord.com/channels/@me/920267778190086205/1192667255406866563

### Screenshots/Video


https://github.com/ecency/ecency-mobile/assets/48380998/0c7d5d98-59b2-4b9d-889c-e67c2d5dcf91

feruzm commented on src/screens/settings/container/settingsContainer.tsx:
```diff
@@ -72,6 +72,17 @@ import ROUTES from '../../../constants/routeNames';
+  'https://rpc.ausbit.dev',
+  'https://api.openhive.network',
+  'https://api.hivekings.com',
+];
```
Please remove pharesim.me and hivekings.com from list and add these

https://techcoderx.com
https://hived.emre.sh
https://hive-api.arcange.eu

@aliseyalvi

noumantahir replied:
@aliseyalvi how about we make it a react-query with initialValue coming from SERVER_LIST constant, as soon api data is fetched, it will be cached for later use, hence fixing the glitch is data miss-match together.

aliseyalvi commented:
@noumantahir 

1. added react query for fetching nodes list
2. updated nodes endpoint
3. updated local copy of server nodes list as @feruzm suggested [here](https://github.com/ecency/ecency-mobile/pull/2813#discussion_r1443642215)

noumantahir reviewed (approved):
lgtm
